src/utils/createProxy.ts

```diff
@@ -44,21 +44,21 @@ function optionsToProxyHandler<T extends Record<string, any>>(
     }
   }
 
-  handler.set = function (target, propertyName, nextValue, receiver) {
+  handler.set = function (target, propertyName, nextValue) {
     const next = () => {
-      const propertySource = findPropertySource(target, propertyName)
-      if (propertySource === null) return false
-
+      const propertySource = findPropertySource(target, propertyName) || target
```

Comment:
The actual fix: `target` is now considered a fallback for `propertySource` if setting properties that don't exist on `target` or its prototype.